modules/core/src/main/scala/Json.scala

```diff
@@ -4,34 +4,40 @@ package schema
 
 import Liskov._
 
-trait JsonModule extends SchemaModule {
+object Json {
   type JSON = String
 
   type Encoder[A] = A => JSON
 
+}
+
+trait JsonModule[R <: Realisation] extends SchemaModule[R] {
+  import Json._
   import Schema._
 
   implicit final def algebra(
-    implicit primNT: Prim ~> Encoder,
-    fieldLabel: ProductTermId <~< String,
-    branchLabel: SumTermId <~< String
-  ): HAlgebra[Schema, Encoder] = new (Schema[Encoder, ?] ~> Encoder) {
+    implicit primNT: R#Prim ~> Encoder,
```

Comment:
I am concerned about the use of `#` as it is slotted for removal. It is a weird situation I do not understand why this works https://github.com/GrafBlutwurst/scalaz-schema/blob/context-free-algebras/modules/core/src/main/scala/ContextFree.scala but the same pattern doesn't work here. 

I did try it out if you `_` the first type param in `IsoSchema` it complains that the types for the pattern don't match up because it expects a `Schema[Encoder, A]` 

If you put in `Encoder` for the first type param you get an erasure warning that due to the compiler flags turns out to be fatal.

I would like to see if we could come up with a solution that doesn't involve `#`. However I have no better idea how to solve this right now.